Build error on Windows: cannot convert argument 2 fr  om 'Napi::Value [1]' to 'const napi_value *' on tclbinding.cpp

Node: v16.14.0(64)
node-gyp: v10.2.0
Operating steps:
1. Download the node-tcl source code
2. Run `npm install bindings`
3. Run `node-gyp rebuild`
4. Return build error, : 
error C2664: 'Napi::Value Napi::Function::Call(napi_value,size_t,const napi_value *) const': cannot convert argument 2 fro
m 'Napi::Value' to 'const napi_value *'

Can you help to resolve this issue or give some suggestion for that? Thank you in advance! 

Below is the detail error sreenshot:
![image](https://github.com/user-attachments/assets/35ddc25b-7fb9-463d-b5f6-c4f4870feee6)

--- a/.github/workflows/build.yaml
+++ b/.github/workflows/build.yaml
@@ -14,15 +14,16 @@ jobs:
     strategy:
       matrix:
         node:
-          - "12"
-          - "14"
           - "16"
+          - "18"
+          - "20"
+          - "22"
     name: Node ${{ matrix.node }}
     steps:
       - name: Install dependencies
         run: sudo apt-get install -y tcl-dev
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node }}
           cache: npm

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -8,14 +8,14 @@ jobs:
     steps:
       - name: Install dependencies
         run: sudo apt-get install -y tcl-dev
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
         with:
           node-version: lts/*
           cache: npm
       - run: npm install
       - run: npm run lint
       - run: npm run coverage:ci
-      - uses: codecov/codecov-action@v2
+      - uses: codecov/codecov-action@v4
         with:
           token: ${{ secrets.CODECOV_TOKEN }}

--- a/README.md
+++ b/README.md
@@ -1,10 +1,8 @@
 node-tcl
 ========
 
-[![Join the chat at https://gitter.im/nukedzn/node-tcl](https://nukedzn.github.io/badges/gitter.svg)](https://gitter.im/nukedzn/node-tcl?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
-
 [![npm version](https://img.shields.io/npm/v/tcl.svg)](https://www.npmjs.com/package/tcl)
-[![Build Status](https://travis-ci.org/nukedzn/node-tcl.svg)](https://travis-ci.org/nukedzn/node-tcl)
+[![build](https://github.com/nukedzn/node-tcl/actions/workflows/build.yaml/badge.svg?branch=master)](https://github.com/nukedzn/node-tcl/actions/workflows/build.yaml)
 [![codecov](https://codecov.io/gh/nukedzn/node-tcl/branch/master/graph/badge.svg?token=AAhVnMCHcn)](https://codecov.io/gh/nukedzn/node-tcl)
 
 Provides Node.js bindings to execute Tcl commands using a native Tcl Interpreter.

--- a/binding.gyp
+++ b/binding.gyp
@@ -41,7 +41,6 @@
 			'include_dirs': [
 				'<!(. <(tclconfig) && echo ${TCL_INCLUDE_SPEC} | sed s/-I//g)',
 				'<!(node -p "require(\'node-addon-api\').include_dir")'
-				               
 			],
 			'dependencies': ["<!(node -p \"require('node-addon-api').gyp\")"],
 			'link_settings': {

--- a/src/tclbinding.cpp
+++ b/src/tclbinding.cpp
@@ -58,16 +58,15 @@ Napi::Value TclBinding::cmd( const Napi::CallbackInfo& info ) {
 		return env.Undefined();
 	}
 
+#ifdef HAS_TCL_THREADS
+	// schedule an async task
 	std::string cmd = info[0].As<Napi::String>().Utf8Value();
 	Napi::Function callback = info[1].As<Napi::Function>();
 
-#ifdef HAS_TCL_THREADS
-	// schedule an async task
 	TclWorker* tw = new TclWorker(callback, cmd.c_str());
 	tw->Queue();
 #else
-	Napi::Value argv[] = {Napi::String::New(env, MSG_NO_TCL_THREADS)};
-	callback->Call( 1, argv );
+	Napi::Error::New(env, MSG_NO_TCL_THREADS).ThrowAsJavaScriptException();
 #endif
 
 	return env.Undefined();
@@ -127,20 +126,21 @@ Napi::Value TclBinding::queue( const Napi::CallbackInfo& info ) {
 		return env.Undefined();
 	}
 
-	std::string cmd = info[0].As<Napi::String>().Utf8Value();
-	Napi::Function callback = info[1].As<Napi::Function>();
+
 
 #if defined(HAS_CXX11) && defined(HAS_TCL_THREADS)
 	if ( this->_tasks == nullptr ) {
 		this->_tasks = new TaskRunner();
 	}
 
+	std::string cmd = info[0].As<Napi::String>().Utf8Value();
+	Napi::Function callback = info[1].As<Napi::Function>();
+
 	// queue the task
 	this->_tasks->queue( callback, cmd.c_str() );
 
 #else
-	Napi::Value argv[] = {Napi::String::New(env, MSG_NO_THREAD_SUPPORT)};
-	callback->Call( 1, argv );
+	Napi::Error::New(env, MSG_NO_TCL_THREADS).ThrowAsJavaScriptException();
 #endif
 
 	return env.Undefined();

--- a/src/tclbinding.cpp
+++ b/src/tclbinding.cpp
@@ -70,7 +70,6 @@ Napi::Value TclBinding::cmd( const Napi::CallbackInfo& info ) {
 #endif
 
 	return env.Undefined();
-
 }
 
 
@@ -208,8 +207,8 @@ Napi::Value TclBinding::proc( const Napi::CallbackInfo& info ) {
 
 	CmdDef *cmdRec = new CmdDef(env);
 	cmdRec->funcRef = Napi::Persistent(info[1].As<Napi::Function>());
-	
-	Tcl_Command code = Tcl_CreateCommand(this->_interp, cmd.c_str(), 
+
+	Tcl_Command code = Tcl_CreateCommand(this->_interp, cmd.c_str(),
 		[](ClientData clientData, Tcl_Interp *ti, int argc, const char *argv[]) {
 			CmdDef *cmdData = (CmdDef *)clientData;
 			Napi::Env env = cmdData->env;

